src/features/hotspots/root/ShortcutNav.tsx

```diff
@@ -216,58 +220,90 @@ const ShortcutNav = ({
     scroll(2, true)
   }, [data.length, scroll, sizes])
 
+  useEffect(() => {
+    if (getItemId(propsItem) === internalItem.id) {
+      return
+    }
+
+    const index = data.findIndex(
+      (item) => getItemId(item) === getItemId(propsItem),
+    )
+
+    if (index === -1) return
+
+    // The selected item was changed from the outside world
+    // Track it and scroll to it
+    setInternalItem({
+      index,
+      id: getItemId(propsItem),
+    })
+
+    scroll(index)
+  }, [data, scroll, propsItem, internalItem.id])
+
   const handleItemSelected = useCallback(
     (item: GlobalOpt | Hotspot) => {
-      setSelectedIndex(data.indexOf(item))
+      if (getItemId(item) === getItemId(propsItem)) {
+        return
+      }
+      setInternalItem({
+        index: data.indexOf(item),
+        id: getItemId(item),
+      })
       onItemSelected(item)
     },
-    [data, onItemSelected],
+    [data, onItemSelected, propsItem],
   )
 
   useEffect(() => {
-    // TODO: There are some alignment issues when follow/unfollow
-    // Need to rethink how to handle this for a future release
+    // if there's a newly followed hotspot, wait for sizes to update then scroll to it
+    if (!hasFollowHotspotChange.current) return
+
+    const allSizesFound = data.every((d) => !!sizes[getItemId(d)])
+    if (!allSizesFound) return
+
+    const nextIndex = data.findIndex((d) => isSelected(d, propsItem))
+    if (nextIndex === -1) return
+
+    hasFollowHotspotChange.current = false
+    scroll(nextIndex)
+  }, [data, isSelected, scroll, propsItem, sizes])
+
+  useEffect(() => {
     if (prevFollowed && followedHotspots.length !== prevFollowed.length) {
+      // remove this item from the size list
+      if (internalItem.index >= GLOBAL_OPTS.length) {
+        let addressToRemove = ''
+        if (propsItem) {
+          addressToRemove = getItemId(propsItem)
+        }
+        sizes[addressToRemove] = null
```

Comment:
so if `!propsItem`, then this is going to execute as `sizes[''] = null`. Is that the desired outcome?

Reply:
No, good catch. I removed the else block I had and didn't adjust this.